src/transformStateWithClones.js

```diff
@@ -1,13 +1,27 @@
 'use strict';
 
-/**
- * @param {Object} state
- * @param {Object[]} actions
- *
- * @return {Object[]}
- */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arrayOfObjects = [];
+  let newObject = { ...state };
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      for (const key in action.extraData) {
```

Comment:
Consider using Object.assign instead of looping through every key in the object to improve the performance